CONTRIBUTING.md

```diff
@@ -63,6 +63,24 @@ https://github.com/elastic/logstash/labels/low%20hanging%20fruit
 
 Using IntelliJ? See a detailed getting started guide [here](https://docs.google.com/document/d/1kqunARvYMrlfTEOgMpYHig0U-ZqCcMJfhvTtGt09iZg/pub).
 
+## Breaking Changes
+
+When introducing new behaviour, we favor implementing it in a non-breaking way that users can opt into, meaning users' existing configurations continue to work as they expect them to after upgrading Logstash.
+
+But sometimes it is necessary to introduce "breaking changes", whether that is removing an option that doesn't make sense anymore, changing the default value of a setting, or reimplementing a major component differently for performance or safety reasons.
+
+When we do so, we need to acknowledge the work we are placing on the rest of our users the next time they upgrade, and work to ensure that they can upgrade confidently.
+
+Where possible, we:
+ 1. first implement new behaviour in a way that users can explicitly opt into (MINOR),
+ 2. introduce deprecation warnings when old behaviour is used, communicating the pending change (MINOR, potentially along-side #1),
+ 3. change the default to be new behaviour, optionally allowing users to opt out in favor of the old behaviour (MAJOR), and eventually
+ 4. remove the old behaviour's implementation from the code-base (MAJOR, potentially along-side #3).
+
+If you request or submit a change that is flagged as a "breaking change", we have several paths forward:
```

Comment:
not confident on the voice of this line, especially how it aligns with the rest of the document. I want to call out that a breaking change isn't the end of the road (without using idioms like "end of the road"), and that contributions with breaking changes aren't just flat-out being rejected.